private/bufpkg/bufanalysis/print.go

```diff
@@ -216,12 +213,25 @@ func newExternalFileAnnotation(f FileAnnotation) externalFileAnnotation {
 	if f.FileInfo() != nil {
 		path = f.FileInfo().ExternalPath()
 	}
+	startLine := f.StartLine()
+	startColumn := f.StartColumn()
+	endLine := f.EndLine()
+	endColumn := f.EndColumn()
+	// If path is empty, we don't output anything
+	// This may just be exacerbating the problem in https://github.com/bufbuild/buf/issues/1930
+	// but it seems right, but not very reasoned
+	if path != "" {
```

Comment:
This would be true if `FileInfo#ExternalPath()` returns `""`. Looking at the implementations, it doesn't seem like this should be the case as `ExternalPath` falls back to `Path`, which is always normalized, validated, and non-empty. Not to mention, the resulting annotation is also useless. I.e., if `path` is empty here, we have a bug. 

Either way, we don't need to guard this block.

Reply:
removed

Reply:
generally, i want to do defensive programming in these scenarios, but normalized paths are a case in the codebase where we assume standard conventions